### ISSUE TITLE
Usage on AWS Lambda with Oauth

### ISSUE THREAD

**Issue by andrewdoro** [ybd-project/ytdl-core#37]
## Describe the bug
I am trying to use this library on AWS Lambda, serverless. I've added the basic oauth configuration. It worked for an hour and now I am getting this error. "Failed to refresh access token" is this happening because of a wrong refresh token value ?. I just copied the one from the generate-own script.

```tsx
import { YtdlCore } from '@ybd-project/ytdl-core/serverless';


const oauth2Credentials = {
  accessToken: Resource.YOUTUBE_ACCESS_TOKEN.value,
  refreshToken: Resource.YOUTUBE_REFRESH_TOKEN.value,
  expiryDate: Resource.YOUTUBE_EXPIRY_DATE.value,
  clientData: {
    clientId: Resource.YOUTUBE_CLIENT_ID.value,
    clientSecret: Resource.YOUTUBE_CLIENT_SECRET.value,
  },
};

  const ytld = new YtdlCore({
      quality: 'lowestaudio',
      filter: 'audioonly',
      hl: 'en',
      gl: 'US',

      disableDefaultClients: true,
      disablePoTokenAutoGeneration: true,
      disableInitialSetup: true,
      parsesHLSFormat: false,
      noUpdate: true,
      logDisplay: ['warning', 'error'],
      clients: ['mweb', 'web'],

      oauth2Credentials: oauth2Credentials,
      html5Player: {
        useRetrievedFunctionsFromGithub: true,
      },
    });

    const info = await ytld.getBasicInfo(event.url, { oauth2Credentials });

    if (info.videoDetails.lengthSeconds > 3600) {
      // 1 hour limit
      throw new Error('Video is too long. Maximum duration is 1 hour.');
    }

    const stream = await ytld.download(event.url, { oauth2Credentials });
```

## Error Details (Log)
[  ERROR  ]: Failed to refresh access token: 401

## Environment

- Video URL: https://www.youtube.com/watch?v=lFlu60qs7_4
- @ybd-project/ytdl-core version: 6.0.8
- Node.js version: 20
- Operating system: Linux

**Comment by andrewdoro** [comment]
It was a wrong environment variable on my side. It's working now. Also I was running the refresh token script and I got this message. Also because I am using serverless is the new accessToken save in any place, or is there an option to do that. Right now I think it will always use the refresh token.

| Important: The uniquely generated tokens may become unavailable if they are not updated for 2~3 days.

Does this mean that the refresh token can become expired?

**Comment by 510ren** [comment]
> | Important: The uniquely generated tokens may become unavailable if they are not updated for 2~3 days.
> 
> Does this mean that the refresh token can become expired?

Normally, tokens that are generated independently should be refreshed every day or so due to their short expiration time. Although we have not experimented in detail, we have determined this because the value of `expiryDate` is one day later.

**Comment by 510ren** [comment]
> It was a wrong environment variable on my side. It's working now. Also I was running the refresh token script and I got this message. Also because I am using serverless is the new accessToken save in any place, or is there an option to do that. Right now I think it will always use the refresh token.

If you are using serverless, the cache would be placed in the temp folder. If you need the ability to retrieve it, this will be added in v6.0.9.

**Comment by andrewdoro** [comment]
Yeah the idea, is to be able to get the new tokens, and save them in a bucket/db. On serverless, the values from the temp folder won't be persisted. 

Thank you for your work

**Comment by andrewdoro** [comment]
But it's still kinda painful that I have to redo the oauth sign in flow again when the refresh token expires. Are there any other solution that don't require manual handling of these. My use case is to just download a video when the transcript is not available and process it's audio with Whisper.

**Comment by 510ren** [comment]
In case you are wondering, OAuth2 tokens are automatically renewed when they expire (after the `expiryDate` date and time).
Since the cache (temporary) of the temp folder is not valid for serverless, should users be able to specify their own caching mechanism?
